tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.td

```diff
@@ -313,4 +313,58 @@ def InlineReductionInitializerOp : Op<Transform_Dialect, "disc.inline_reduction_
   let cppNamespace = "::mlir::disc_ral::transform_dialect";
 }
 
+def DecomposeVectorsOp : Op<Transform_Dialect, "disc.decompose_vectors",
+    [FunctionalStyleTransformOpTrait,
+     MemoryEffectsOpInterface,
+     TransformEachOpTrait,
+     TransformOpInterface]> {
+  let description = [{
+    Decompose vector ops and related vector transform_read/write ops into fine-grained-size vector ops.
+
+    Without this decomposition, llvm backend will mis-allocate vector register for some vector ops
+    with specific shape (e.g. 8x12xf32 failed to map to hardware register). This is a workaround for this.
```

Comment:
Is this a bug of LLVM?

Reply:
It's not a bug of LLVM. It's just not supported well by the llvm aarch64 backend.